app/angular/src/client/preview/angular-beta/DocsRenderer.ts

```diff
@@ -26,6 +26,15 @@ export class DocsRenderer extends AbstractRenderer {
       await DocsRenderer.resetPlatformBrowserDynamic();
     });
 
+    /**
+     * Destroy and recreate the PlatformBrowserDynamic of angular
+     * when doc re render. Allows to call ngOnDestroy of angular
+     * for previous component
+     */
+    channel.once(Events.DOCS_RENDERED, async () => {
```

Comment:
Does `DOCS_RENDERED` mean "after docs have rendered"? 
Why do we need to reset everything after the rendering? Or is the Event name just misleading?

Reply:
when we are on the page of docs and that we make a change in the html for example the doc is rendered again. and it is thus necessary to destroy the previous component 

tested with `/story/basics-component-with-ngondestroy--simple-component` 

we see clearly the problem after several HMR in console